File: javalin5/javalin-http2-example/src/main/java/app/JavalinHttp2ExampleApp.java
package app;

import io.javalin.Javalin;
import io.javalin.community.ssl.SSLPlugin;
import io.javalin.http.staticfiles.Location;

public class JavalinHttp2ExampleApp {

    public static void main(String[] args) {

        Javalin app = Javalin.create(config -> {

            config.plugins.register(new SSLPlugin(sslConfig -> {
                sslConfig.keystoreFromClasspath("keystore.jks", "password"); // The ssl plugin will enable HTTP/2 by default
                sslConfig.insecurePort = 8080;
                sslConfig.securePort = 8443;
            }));

            config.staticFiles.add("/public", Location.CLASSPATH);
        }).start();

        app.get("/", ctx -> ctx.result("Hello World"));

    }

}

File: javalin5/javalin-http2-example/src/main/kotlin/app/JavalinHttp2ExampleApp.kt
package app

import io.javalin.Javalin
import io.javalin.community.ssl.SSLPlugin
import io.javalin.http.staticfiles.Location

fun main() {

    val app = Javalin.create {
        it.plugins.register(SSLPlugin {
            it.keystoreFromClasspath("keystore.jks", "password")
            it.insecurePort = 8080
            it.securePort = 8443
        })
        it.staticFiles.add("/public", Location.CLASSPATH)
    }.start()

    app.get("/") { it.result("Hello World") }

}

